epoxy-processortest/src/test/java/com/airbnb/epoxy/ModelProcessorTest.java

```diff
@@ -498,4 +487,179 @@ public void modelWithViewClickListener() {
+        .forResource("ModelWithPrivateFieldWithGetterWithParams.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithPrivateAttributeWithGetterWithWrongReturnTypeFails() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithPrivateFieldWithGetterWithWrongReturnType.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithPrivateAttributeWithPrivateSetterFails() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithPrivateFieldWithPrivateSetter.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithPrivateAttributeWithStaticSetterFails() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithPrivateFieldWithStaticSetter.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithPrivateAttributeWithGetterWithoutParamsFails() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithPrivateFieldWithSettterWithoutParams.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithPrivateAttributeWithSetterWithWrongReturnTypeFails() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithPrivateFieldWithSetterWithWrongParamType.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .failsToCompile()
+        .withErrorContaining("private fields without proper getter and setter");
+  }
+
+  @Test
+  public void testModelWithAllPrivateFieldTypes() {
+    JavaFileObject model = JavaFileObjects
+        .forResource("ModelWithAllPrivateFieldTypes.java");
+
+    JavaFileObject generatedModel = JavaFileObjects
+        .forResource("ModelWithAllPrivateFieldTypes_.java");
+
+    assert_().about(javaSource())
+        .that(model)
+        .processedWith(new EpoxyProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(generatedModel);
+  }
+
+  @Test
+  public void modelWithViewPrivateClickListener() {
```

Comment:
No problem, I am jus trying to cover every possible case. This is the purpose of tests after all :)